File: solutions/java/src/snakeandladdergame/GameManager.java
package snakeandladdergame;

import java.util.ArrayList;
import java.util.List;

public class GameManager {
    private static GameManager instance;
    private final List<SnakeAndLadderGame> games;

    private GameManager() {
        games = new ArrayList<>();
    }

    public static synchronized GameManager getInstance() {
        if (instance == null) {
            instance = new GameManager();
        }
        return instance;
    }

    public void startNewGame(List<String> playerNames) {
        SnakeAndLadderGame game = new SnakeAndLadderGame(playerNames);
        games.add(game);
        new Thread(() -> game.play()).start();
    }
}

File: solutions/golang/snakeandladdergame/game_manager.go
package snakeandladdergame

import "sync"

type GameManager struct {
	Games []*SnakeAndLadderGame
}

var instance *GameManager

func GetGameManager() *GameManager {
	if instance == nil {
		instance = &GameManager{
			Games: []*SnakeAndLadderGame{},
		}
	}
	return instance
}

func (gm *GameManager) StartNewGame(wg *sync.WaitGroup, playerNames []string) {
	game := NewSnakeAndLadderGame(playerNames)
	gm.Games = append(gm.Games, game)
	wg.Add(1)
	go game.Play(wg)
}
